lib/include/up-client-zenoh-cpp/rpc/zenohRpcClient.h

```diff
@@ -25,70 +25,122 @@
 #ifndef _ZENOH_RPC_CLIENT_H_
 #define _ZENOH_RPC_CLIENT_H_
 
+#include <condition_variable>
 #include <up-cpp/rpc/RpcClient.h>
 #include <up-cpp/utils/ThreadPool.h>
+#include <up-cpp/transport/datamodel/UPayload.h>
 #include <up-core-api/ustatus.pb.h>
 #include <up-core-api/uri.pb.h>
 #include <zenoh.h>
 
-using namespace std;
-using namespace uprotocol::utransport;
-using namespace uprotocol::utils;
-using namespace uprotocol::v1;
-
-class ZenohRpcClient : public RpcClient {
-
-    public:
-
-        ZenohRpcClient(const ZenohRpcClient&) = delete;
-        ZenohRpcClient& operator=(const ZenohRpcClient&) = delete;
-
-        /**
-        * The API provides an instance of the zenoh RPC client
-        * @return instance of ZenohUTransport
-        */
-        static ZenohRpcClient& instance(void) noexcept;
-
-        /**
-        * init the zenohRpcClient 
-        * @return Returns OK on SUCCESS and ERROR on failure
-        */
-        UStatus init() noexcept;
-
-        /**
-        * Terminates the zenoh RPC client  - the API should be called by any class that called init
-        * @return Returns OK on SUCCESS and ERROR on failure
-        */
-        UStatus term() noexcept; 
-
-        /**
-        * Support for RPC method invocation.
-        * @param topic topic of the method to be invoked (i.e. the name of the API we are calling).
-        * @param payload The request message to be sent to the server.
-        * @param attributes Metadata for the method invocation (i.e. priority, timeout, etc.)
-        * @return Returns the CompletableFuture with the result or exception.
-        */
-        std::future<UPayload> invokeMethod(const UUri &uri, 
-                                           const UPayload &payload, 
-                                           const UAttributes &attributes) noexcept;
-    private:
-
-        ZenohRpcClient() {}
-
-        static UPayload handleReply(z_owned_reply_channel_t *channel);
-        
-        /* zenoh session handle*/
-        z_owned_session_t session_;
-        /* how many times uTransport was initialized*/
-        atomic_uint32_t refCount_ = 0;
-        
-        std::mutex mutex_;
-        std::shared_ptr<ThreadPool> threadPool_;
-
-        static constexpr auto requestTimeoutMs_ = 5000;
-        static constexpr auto queueSize_ = size_t(20);
-        static constexpr auto maxNumOfCuncurrentRequests_ = size_t(2);
-
-};
+namespace uprotocol::rpc {
+
+    struct ZenohRpcClientConfig
+    {
+        size_t maxQueueSize;
+        size_t maxConcurrentRequests;
+    };
+
+    class ZenohRpcClient : public uprotocol::rpc::RpcClient {
```

Comment:
I think I recall the direction chosen was to unify into a single e.g. `UPClientZenoh` that would implement both `UTransport` and `RpcClient`. Kinda sanity checking myself.

Maybe again -- handling in another PR?

Reply:
I believe that is being done in [this PR on Michael's fork](https://github.com/MelamudMichael/up-client-zenoh-cpp/pull/7).

Reply:
Yes @PLeVasseur  this is the plan, but would still have two separate classes that hold the implementation for RPC and transport